### PR TITLE
[Security Content] Added Compatibility note to all IGs

### PR DESCRIPTION
**Pull request by Aegrah**
## Summary
Added the following note:

`> This investigation guide uses [placeholder fields](https://www.elastic.co/guide/en/security/current/osquery-placeholder-fields.html) to dynamically pass alert data into Osquery queries. Placeholder fields were introduced in Elastic Stack version 8.7.0. If you're using Elastic Stack version 8.6.0 or earlier, you'll need to manually adjust this investigation guide's queries to ensure they properly run.`

To the following IGs:

- Potential Persistence Through init.d Detected
- Potential Linux Backdoor User Account Creation
- Linux Group Creation
- Potential Remote Code Execution via Web Server
- Linux User Account Creation
- Linux User Added to Privileged Group
- Potential Persistence Through MOTD File Creation Detected
- Suspicious Process Spawned from MOTD Detected
- Potential Persistence Through Run Control Detected
- New Systemd Timer Created

### CODE REVIEW

**Comment by w0rk3r** [comment]
@nastasha-solomon @joepeeples can you take a look at this one please?

**Comment by nastasha-solomon** [comment]
@w0rk3r - sure! What's the deadline for getting this reviewed? Does it need to be completed before 8.9 GAs?

**Comment by w0rk3r** [comment]
Hey @nastasha-solomon, this is for 8.9, but only the Osquery note needs review here

File: rules/linux/persistence_init_d_file_creation.toml [hunk 1]
```diff
@@ -68,7 +68,7 @@ This rule looks for the creation of new files within the `/etc/init.d/` director
 
 > **Note**:
 > This investigation guide uses the [Osquery Markdown Plugin](https://www.elastic.co/guide/en/security/master/invest-guide-run-osquery.html) introduced in Elastic Stack version 8.5.0. Older Elastic Stack versions will display unrendered Markdown in this guide.
-
+> This investigation guide leverages functionality that allows the usage of variables within the OSQuery queries. These variables will be directly populated by the corresponding values within the detection alert at run time. This feature has been introduced in Elastic Stack version 8.7.0. Older Elastic Stack versions will not be able to leverage this feature, and will require manual adjustments to run the query.
```
Comment (nastasha-solomon) [inline rules/linux/persistence_init_d_file_creation.toml:71]: Just some slight modifications to keep the language consistent with our [feature docs](https://www.elastic.co/guide/en/security/current/osquery-placeholder-fields.html) and the format of the previous note:

```suggestion
> This investigation guide uses [placeholder fields](https://www.elastic.co/guide/en/security/current/osquery-placeholder-fields.html) to dynamically pass alert data into Osquery queries. Placeholder fields were introduced in Elastic Stack version 8.7.0. If you're using Elastic Stack version 8.6.0 or earlier, you'll need to manually adjust this investigation guide's queries to ensure they properly run.   
```

**Review by nastasha-solomon — COMMENTED** [review]
Thanks for adding this note! If you accept my edits, they'll apply to all the notes in this PR.

**Comment by Aegrah** [comment]
Implemented the feedback, it's in all of the rules now. Thanks!

**Review by nastasha-solomon — APPROVED** [review]

File: rules/linux/persistence_init_d_file_creation.toml [hunk 2]
```diff
@@ -68,7 +68,7 @@ This rule looks for the creation of new files within the `/etc/init.d/` director
 
 > **Note**:
 > This investigation guide uses the [Osquery Markdown Plugin](https://www.elastic.co/guide/en/security/master/invest-guide-run-osquery.html) introduced in Elastic Stack version 8.5.0. Older Elastic Stack versions will display unrendered Markdown in this guide.
-
+> This investigation guide uses [placeholder fields](https://www.elastic.co/guide/en/security/current/osquery-placeholder-fields.html) to dynamically pass alert data into Osquery queries. Placeholder fields were introduced in Elastic Stack version 8.7.0. If you're using Elastic Stack version 8.6.0 or earlier, you'll need to manually adjust this investigation guide's queries to ensure they properly run.
```
Comment (w0rk3r) [inline rules/linux/persistence_init_d_file_creation.toml:71]:
```suggestion
> This investigation guide uses [placeholder fields](https://www.elastic.co/guide/en/security/current/osquery-placeholder-fields.html) to dynamically pass alert data into Osquery queries. Placeholder fields were introduced in Elastic Stack version 8.7.0. If you're using Elastic Stack version 8.6.0 or earlier, you'll need to manually adjust this investigation guide's queries to ensure they properly run.
```

**Review by w0rk3r — APPROVED** [review]
Nice! Thanks @Aegrah @nastasha-solomon

**Pull request merged**